outage/2024-10-09-ota-lite.md

```diff
@@ -0,0 +1,57 @@
+## OTA Lite Outage
+
+OTA-lite was unavailble from 20:13UTC to 20:35UTC. During this time things
+like publishing TUF Targets and managing devices (from user facing APIs)
+was not possible. The device-gateway remained up during this outage.
+
+### What Happened
+
+On the previous day, 2024-10-08, we performed a task where we periodically
+rotate secure keys and tokens used by our backend. After rotating keys, we
+restarted each container affected by the key rotation to make sure they
+continuted to function properly. We also deleted some old service accounts
+and tokens no longer used by our system.
+
+We went to bed and slept comfortably.
+
+Later the next day OTA lite would not start up correctly after we tried
+to deploy a very minor change to it. We tried to rollback the change only
+to discover it couldn't run either.
+
+The failure logs pointed at the fact we were using a key that didn't exist.
+
+The missing key was for a minor part of the system where we keep backups of
+each version of Targets.json a customer publishes. It's not stricly required,
+so the quick fix was to comment out this code to give us breathing room.
+
+The root cause wound up being a combination of a bad/naive Grep expression
+combined with a bad variable naming. We were referencing a Kubernetes secret
+in a Pod with:
+```
+      - name: gce-creds
+        secret:
+          defaultMode: 420
+          secretName: ota-api-artifacts
+```
```

Comment:
Not sure if we should disclose our secret names to the general public... as that makes them, well, less secret.
That one sentence (the first one) is enough IMHO.

Feel free to add 1-2 sentences of some detail if your see fit; but, please, don't disclose our infrastructure internals.

Reply:
We can drop it but not for security reasons. If we have a threat vector where knowing the name of a secret compromises us - then we've failed.

Less text is often good. That should be the reason we drop this.